rust_dev_preview/examples/rds/src/aurora.rs

```diff
@@ -0,0 +1,664 @@
+                Parameter::builder()
+                    .parameter_name("auto_increment_increment")
+                    .parameter_value(format!("{increment}"))
+                    .apply_method(aws_sdk_rds::types::ApplyMethod::Immediate)
+                    .build(),
+            )
+            .send()
+            .await;
+
+        if let Err(error) = modify_db_cluster_parameter_group {
+            return Err(ScenarioError::new(
+                "Failed to modify cluster parameter group",
+                &error,
+            ));
+        }
+
+        Ok(())
+    }
+
+    // Get a list of allowed engine versions. rds.DescribeDbEngineVersions(Engine='aurora-mysql', DBParameterGroupFamily=<the family used to create your parameter group in step 2>)
+    // Create an Aurora DB cluster database cluster that contains a MySql database and uses the parameter group you created.
+    // Wait for DB cluster to be ready. Call rds.DescribeDBClusters and check for Status == 'available'.
+    // Get a list of instance classes available for the selected engine and engine version. rds.DescribeOrderableDbInstanceOptions(Engine='mysql', EngineVersion=).
+
+    // Create a database instance in the cluster.
+    // Wait for DB instance to be ready. Call rds.DescribeDbInstances and check for DBInstanceStatus == 'available'.
+    pub async fn start_cluster_and_instance(&mut self) -> Result<(), ScenarioError> {
+        if self.password.is_none() {
+            return Err(ScenarioError::with(
+                "Must set Secret Password before starting a cluster",
+            ));
+        }
+        let create_db_cluster = self
+            .client
+            .create_db_cluster()
+            .db_cluster_identifier(DB_CLUSTER_IDENTIFIER)
+            .db_cluster_parameter_group_name(DB_CLUSTER_PARAMETER_GROUP_NAME)
+            .engine(DB_ENGINE)
+            .set_engine_version(self.engine_version.clone())
+            .set_master_username(self.username.clone())
+            .master_user_password(self.password.as_ref().unwrap().expose_secret())
+            .send()
+            .await;
+        if let Err(err) = create_db_cluster {
+            return Err(ScenarioError::new(
+                "Failed to create DB Cluster with cluster group",
+                &err,
+            ));
+        }
+
+        self.db_cluster_identifier = create_db_cluster
+            .unwrap()
+            .db_cluster
+            .and_then(|c| c.db_cluster_identifier);
+
+        if self.db_cluster_identifier.is_none() {
+            return Err(ScenarioError::with("Created DB Cluster missing Identifier"));
+        }
+
+        info!(
+            "Started a db cluster: {}",
+            self.db_cluster_identifier
+                .clone()
+                .unwrap_or("Missing ARN".to_string())
+        );
+
+        let create_db_instance = self
+            .client
+            .create_db_instance()
+            .set_db_cluster_identifier(self.db_cluster_identifier.clone())
```

Comment:
Same question here, I guess it doesn't take a reference?

Reply:
Done.